merlin-sdk/src/main/java/gov/nasa/jpl/aerie/merlin/protocol/model/Task.java

```diff
@@ -20,7 +25,174 @@
    * resource leaks</p>
    *
    * <p>This method <b>shall not</b> be called on this object after invoking {@code #step(Scheduler)};
-   * nor shall {@link #step(Scheduler)} be called after this method.</p>
+   * nor shall {@link #step(Scheduler, Input)} be called after this method.</p>
    */
   default void release() {}
+
+  /**
+   * Perform another task following this one.
+   *
+   * @param <X>
+   *   The type of the output of the suffixed task.
+   * @param suffix
+   *   The task to perform following this one.
+   * @return
+   *   A task performing this task and the suffixed task in sequence.
+   */
+  default <X> Task<Input, X> andThen(final Task<Output, X> suffix) {
+    return Task.compose(this, suffix);
+  }
+
+  /**
+   * Perform another task preceding this one.
+   *
+   * @param <X>
+   *   The type of the output of the prefixed task.
+   * @param prefix
+   *   The task to perform preceding this one.
+   * @return
+   *   A task performing the prefixed task and this task in sequence.
+   */
+  default <X> Task<X, Output> butFirst(final Task<X, Input> prefix) {
+    return Task.compose(prefix, this);
+  }
+
+  /**
+   * Chain an effect-free transformation onto the output from this task.
+   *
+   * @param <X>
+   *   The type of the new output from the output transformation.
+   * @param step
+   *   The transformation to apply over this task's output.
+   * @return
+   *   A task performing this task and the output transformation in sequence.
+   */
+  default <X> Task<Input, X> andThen(final Function<Output, X> step) {
+    return this.andThen(Task.lift(step));
+  }
+
+  /**
+   * Chain an effect-free transformation onto the input to this task.
+   *
+   * @param <X>
+   *   The type of the new input to the input transformation.
+   * @param step
+   *   The transformation to apply over this task's input.
+   * @return
+   *   A task performing the input transformation and this task in sequence.
+   */
+  default <X> Task<X, Output> butFirst(final Function<X, Input> step) {
+    return this.butFirst(Task.lift(step));
+  }
+
+
+  /**
+   * Create a task that performs a pure, side-effect-free function.
+   *
+   * @param <A>
+   *   The input type of the given function.
+   * @param <B>
+   *   The output type of the given function.
+   * @param step
+   *   The pure function to perform as a task.
+   * @return
+   *   A task performing the given pure function.
+   */
+  // LAW: compose(lift(f), lift(g)) === lift(f.andThen(g))
```

Comment:
If I'm following this correctly, the intent behind this comment was:
```suggestion
  // LAW: compose(lift(f), lift(g)) === lift(f).andThen(lift(g)))
```

Reply:
No, the idea is that compose-then-lift should behave the same as lift-then-compose. (In Haskell-y syntax, `fmap g . fmap f == fmap (g . f)`.)